File: Enalotto.java

import java.util.*;
import java.util.Random;

public class Enalotto {

	public static void main(String[] args) {
		TreeSet<Integer> numbers = new TreeSet<Integer>();

		while(numbers.size() < 6) {
			numbers.add(randInt(1, 90));
		}

		int index = 0;
		for (Integer item : numbers) {
			if(item < 10) System.out.print("0");
			System.out.print(item);
			if(index < 5) System.out.print(" - ");
			index++;
		}

		System.out.println("");
	}

	public static int randInt(int min, int max) {
		Random rand = new Random();
		int randomNum = rand.nextInt((max - min) + 1) + min;
		return randomNum;
	}
	
}

File: enalotto.kt

package com.prontobenedict.enalotto

import java.util.*

fun main(args : Array<String>) {
    val numbers: SortedSet<Int> = intArrayOf().toSortedSet()

    while(numbers.count() < 6) {
        numbers.add(randInt(1, 90))
    }

    for ((index, number) in numbers.withIndex()) {
        if(number < 10) print("0")
        print(number)
        if(index < 5) print(" - ")
    }
}

fun randInt(min: Int, max: Int): Int {
    val rand = Random()
    val randomNum = rand.nextInt(max - min + 1) + min
    return randomNum
}
